### PR TITLE
Add queue_time metric in APM traces as tag

### PR DESCRIPTION
**Pull request by drodriguezhdez**
### Requirements for Contributing to this repository

* Fill out the template below. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* The pull request must only fix one issue at the time.
* The pull request must update the test suite to demonstrate the changed functionality.
* After you create the pull request, all status checks must be pass before a maintainer reviews your contribution. For more details, please see [CONTRIBUTING](/CONTRIBUTING.md). 

### What does this PR do?

* Add queue_time metric in APM traces as tag

### Review checklist (to be filled by reviewers)

- [ ] Feature or bug fix MUST have appropriate tests (unit, integration, etc...)
- [ ] PR title must be written as a CHANGELOG entry [(see why)](/CONTRIBUTING.md#pull-request-title)
- [ ] Files changes must correspond to the primary purpose of the PR as described in the title (small unrelated changes should have their own PR)
- [ ] PR must have one `changelog/` label attached. If applicable it should have the `backward-incompatible` label attached.
- [ ] PR should not have `do-not-merge/` label attached.
- [ ] If Applicable, issue must have `kind/` and `severity/` labels attached at least.

### CODE REVIEW

**Review by hithwen — APPROVED** [review]

File: src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java [hunk 1]
```diff
@@ -99,6 +100,11 @@ public void finishBuildTrace(final BuildData buildData, final Run<?,?> run) {
         buildSpan.setTag(prefix + CITags._NUMBER, buildData.getBuildNumber(""));
         buildSpan.setTag(prefix + CITags._URL, buildData.getBuildUrl(""));
 
+        final TimeInQueueAction timeInQueueAction = run.getAction(TimeInQueueAction.class);
+        if(timeInQueueAction != null) {
+            buildSpan.setTag(prefix + CITags._QUEUE_TIME, timeInQueueAction.getSecondsInQueue());
```
Comment (FlorianVeaux) [inline src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java:105]: Not familiar with how the intake will work, but tagging with a float is a dangerous thing for metrics leading to context explosion very fast.
Are we sure the backend can handle float values as tags in an APM context?

**Review by FlorianVeaux — COMMENTED** [review]
Looks great jenkins wise, one comment regarding the tag though as I'm not sure if the APM backend will be able to handle it

File: src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java (hunk 1 above)
Comment (drodriguezhdez) [reply src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java:105]: I'm confused. The tag is based on a `long` value. In this case, the number of seconds calculated as `long` in https://github.com/jenkinsci/datadog-plugin/blob/master/src/main/java/org/datadog/jenkins/plugins/datadog/listeners/DatadogBuildListener.java#L108

Comment (FlorianVeaux) [reply src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java:105]: Sorry I've used "float" for "number" in that context. The point being that the value of the tag will be different for each span

Comment (drodriguezhdez) [reply src/main/java/org/datadog/jenkins/plugins/datadog/traces/DatadogTraceBuildLogic.java:105]: Ah!, It's totally supported by the backend. No worries. It's based on the Opentracing standard.

image

**Review by FlorianVeaux — APPROVED** [review]

**Pull request merged**